lib/Monad_WP/WhileLoopRules.thy

```diff
@@ -562,7 +562,7 @@ lemma whileLoop_terminates_liftE:
    apply clarsimp
   apply (subgoal_tac "case (Inr r) of Inl _ \<Rightarrow> True | Inr r \<Rightarrow>
       whileLoop_terminates (\<lambda>r s. (\<lambda>r s. case r of Inl _ \<Rightarrow> False | Inr v \<Rightarrow> C v s) (Inr r) s)
-           (\<lambda>r. (lift (\<lambda>r. liftE (B r)) (Inr r)) >>= (\<lambda>x. return (theRight x))) r s")
+           (\<lambda>r. (lift (\<lambda>r. liftE (B r)) (Inr r)) >>= (\<lambda>x. return (projr x))) r s")
```

Comment:
confirm: should `theRight` changes be in this PR?

Reply:
They shouldn't, but they aren't :-)

`theRight` is in Lib, and these theories don't see `Lib` any more, so they need to use the primitive concept that people outside l4v would use. Once the other PR is merged, we still need `projr` here, but we'd see `theRight` in goals via the abbreviation declared in Lib.